Gems/ONNX/Code/Source/Clients/Mnist.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Mnist.h"
+
+namespace ONNX
+{
+    template<typename T>
+    static void MNIST::softmax(T& input)
+    {
+        float rowmax = *std::max_element(input.begin(), input.end());
+        std::vector<float> y(input.size());
+        float sum = 0.0f;
+        for (size_t i = 0; i != input.size(); ++i)
+        {
+            sum += y[i] = std::exp(input[i] - rowmax);
+        }
+        for (size_t i = 0; i != input.size(); ++i)
+        {
+            input[i] = y[i] / sum;
+        }
+    }
+
+    std::ptrdiff_t MNIST::GetResult()
```

Comment:
Nit: Somehow the return type feels weird and I'd just use an AZ::u32 here and if needed, convert it from a ptrdiff to an int inside the function

Reply:
Changed the return type here to void. Returning a value here is unnecessary as members are mutated directly.